kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/writer/DhcpServerConfigWriterTest.java

```diff
@@ -347,14 +424,32 @@ private void thenKuraExceptionIsCaught() {
         assertTrue(this.isKuraException);
     }
 
+    /*
+     * Utilities
+     */
+
     @After
-    public void clean() {
-        File dhcpConfigFile = new File("/tmp/dhcpd-eth0.conf");
-        if (dhcpConfigFile.exists()) {
-            dhcpConfigFile.delete();
+    public void cleanUp() throws IOException {
+        dhcpServerMock.close();
+        this.mockFiles.delete();
+        this.mockFiles.create();
+    }
+
+    private class PropertiesBuilder {
+
+        private final Map<String, Object> properties;
+
+        public PropertiesBuilder() {
+            this.properties = new HashMap<>();
         }
-        if (dhcpServerMock != null) {
-            dhcpServerMock.close();
+
+        public PropertiesBuilder with(String name, Object value) {
+            this.properties.put(name, value);
+            return this;
+        }
+
+        public void build() {
+            DhcpServerConfigWriterTest.this.networkProperties = this.properties;
```

Comment:
Even though I really like the use of the builder pattern here, I don't like this. 

I find somewhat surprising that a `build()` method can have side effects on its wrapping class. Is there a better way?